assets/src/components/CustomLink/CustomLink.js

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@wordpress/element';
+import {URLInput} from '@wordpress/block-editor'
+import {TextControl as BaseTextControl} from '@wordpress/components';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+
+const TextControl = withCharacterCounter( BaseTextControl );
+
+export class CustomLink extends Component {
```

Comment:
It could also just have the same name as the WP component it's wrapping (`URLInput`). Sure it's not the exact same component, but it's still an input for a url, so would be nice if it still had that descriptive name. It would also mean you don't need to change the component name everywhere it's used, you only need to change the import to point to the new one and add the new attributes.

You can alias the import of WP's `URLInput` to resolve the name conflict.